docs/software/spack.md

```diff
@@ -0,0 +1,179 @@
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack list
+```
+
+The above command lists all 7000+ packages available for installation with Spack. As a result, this command may take some time to execute.  To search for a specific software by name, you can use `spack list <software name>`. For example: 
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack list fastqc
+fastqc
+==> 1 packages
+```
+
+>For additional information about a specific piece of software (available versions, variants, and dependencies), you can use the `spack info <software name>` command. Note that the 'preferred version' is the default version which will be installed if no alternative version is specified. 
+
+Once you've confirmed that a piece of software is available, you can install it in the active environment using the `spack install --add` command:
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack install --add fastqc
+```
+>Note that the `--add` argument is required to add package specs to an environment. You can separately add specs without installing a package using the `spack add <software name>` command.
+
+This will install the specified software and all dependencies into your environment.
+
+You can also modify the installation commands to meet your installation needs. You can specify which version you'd like installed using the `@` operator:
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack install --add fastqc@0.11.9
+```
+
+#### Installation Locations and Specifics
+
+The root of the Spack installation tree, by default, is `/projects/$USER/software/spack` (this can be changed in `~/config.yaml`). From there, executables can be found in subdirectories depending on the OS and compiler used to install the software. Software will be installed here regardless of whether the software was installed in an environment. 
+
+Spack will always check the installation root prior to installing software, in an environment or otherwise. If the software is not found in the installation root, it will be installed there. If the software is found in the installation root, it will not be installed. When attempting to install in an environment, symbolic links to the installation root will be created in `/projects/$USER/spack/environments/<envname>/.spack-env/view`. This behavior means that once a piece of software is installed, it can be linked to any environment without having to perform the installation again. 
+
+The default cache is located at `/scratch/alpine/$USER/spack/cache`. This directory will contain any previously-installed tarballs and repositories, as well as any miscellaneous cache items.  
+
+### Installing and Using Compilers with Spack
+
+In addition to standard software packages, you can use Spack to install compilers which are not currently available in the Alpine software stack. For example, if you need the newest available version of `gcc`, you can install it into your environment as follows: 
+
+```
+[johndoe@c3cpu-c11-u17-2 ~]$ spack install --add gcc@13.1.0
+```
```

Comment:
```suggestion
In addition to standard software packages, you can use Spack to install compilers which are not currently available in the Alpine software stack. It is highly recommended to __NOT__ install compilers directly within the environment. Instead, we suggest that you first install the compiler outside of the environment. If it is installed directly within an environment, then it will depend on the default compiler within the environment. The following is our recommended way to install a new compiler and use it as the compiler for your environment. 

1. Install the compiler outside of your environment (only needs to be done once):
```code
spack install gcc@13.1.0
```code

2. Obtain the location of the compiler install and save it into a local environmental variable:
```code
gcc_location=$(spack location -i gcc@13.1.0)
```code

3. Create and activate your environment:
```code
spack env create my_test_env
spack env activate my_test_env
```code

4. Remove any compilers that are in the env right now:
```code
spack compiler remove gcc -a
```code
>Note that you can also remove intel compilers using `spack compiler remove intel -a`

5. Add the compiler you installed outside of the environment and install the compiler into the environment:
```code
spack compiler add $gcc_location
spack install --add gcc@13.1.0
```code
>Note that the install of the compiler within the environment is necessary as it will install all dependencies needed for the compiler. 
```

Please check the formatting on the "```code" block, like in my previous comment.